imports/wasi_snapshot_preview1/fs.go

```diff
@@ -604,11 +682,6 @@ func fdReaddirFn(_ context.Context, mod api.Module, params []uint64) Errno {
 		return errno
 	}
 
-	// expect a cookie only if we are continuing a read.
```

Comment:
this needs some explanation. a directory that has already been read, asking for the initial state smells like a bug which would imply buffering a really large directory until the FD is closed right?